### PR TITLE
New package: cross-mips-linux-muslhf-0.24 

### PR DESCRIPTION
**Pull request by hipperson0**
Added cross build profile for MIPS32 BE hard float and added support for the mipshf-musl arch to packages which have arch specfic requirements.

Not sure if the changed packages require revbumps or not.

### CODE REVIEW

**Comment by chneukirchen** [comment]
What's the target hardware for this architecture?

**Comment by hipperson0** [comment]
It's the same as that of mips toolchain but with support for using the floating point unit.

Such as CPUs based on MIPS 24kf or kc like the MIPS Malta board.

File: srcpkgs/qt/template [hunk 1]
```diff
@@ -85,6 +85,10 @@ do_configure() {
 				_opts+=" -arch mips"
 				_spec="mips-linux-musl-c++"
 				;;
+			mipshf-musl)
+				_opts+=" -arch mips"
+				_spec="mips-linux-musl-c++"
```
Comment (pullmoll) [inline srcpkgs/qt/template]: Shouldn't that be `mips-linux-mulshf-c++`?

Comment (hipperson0) [reply srcpkgs/qt/template]: Yes, I missed that. Changing it now

**Comment by pullmoll** [comment]
LGTM. Revbumps are not necessary because AFAICT nothing would change for the existing builders architectures.

**Comment by hipperson0** [comment]
Cool, that's what I thought, just wanted to be sure!

**Comment by pullmoll** [comment]
I updated the musl libc to 1.1.18 and binutils to 2.29.1 before merging. I tested that locally and the binary-bootstrap plus quite a number of packages built [can be found here](http://repo.voidlinux.de/current/).

**Comment by hipperson0** [comment]
Thank you very much pullmoll.

**Pull request closed without merging**